src/server/middleware/HeaderHandler.ts

```diff
@@ -0,0 +1,22 @@
+import { HttpHandler } from '../HttpHandler';
+import type { HttpResponse } from '../HttpResponse';
+
+/**
+ * Handler that sets custom headers on the response.
+ */
+export class HeaderHandler extends HttpHandler {
+  private readonly headers: Record<string, string>;
+
+  // Not supported by Components.js yet
+  // eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
+  public constructor(headers: { [header: string]: string } = {}) {
```

Comment:
What exactly is the issue here? Because components.js does support records now afaik?

Reply:
I didn't even test, I just looked at your code 😂
https://github.com/solid/community-server/blob/6b4088723dcafb0756a34e1ff8b41f86014b8fb4/src/ldp/http/metadata/MappedMetadataWriter.ts#L13

Lemme try to upgrade both.